### PR TITLE
chore: Add CI check for formatting

### PR DESCRIPTION
**Pull request by lavaleri**
*Issue #, if available:*

*Description of changes:* Add check for formatting to CI, and format all code to adhere to google-java-format.

Notes:

- Add https://github.com/coveooss/fmt-maven-plugin to pom
- Add CI to specifically check the format
- Run autoformatter tool in order to pass this new check
- Point submodules to public repos

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

# Check any applicable:
- [ ] Were any files moved? Moving files changes their URL, which breaks all hyperlinks to the files.

### CODE REVIEW

**Comment by seebees** [comment]
I vote that we only have the checks in the CI.

**Comment by lavaleri** [comment]
Used jardiff to determine that there is no bytecode change between a jar produced by this branch and master.

Used `diff` to ensure that javadocs produced by this branch and master have no difference, when stripped of newlines and `<!--Generatedbyjavadoc(11.0.10)onWedMayXXXX:XX:XXPDT2021-->` comments.

**Comment by lavaleri** [comment]
Reconfirmed that there is no difference in bytecode using jardiff between this branch and master.

**Comment by seebees** [comment]
Can we move this to https://github.com/aws/aws-encryption-sdk-java/blob/master/codebuild/compliance.yml?

**Comment by lavaleri** [comment]
@seebees You can to run the format check in the same build as the compliance check?

**Comment by seebees** [comment]
That is what I was thinking on the JS side, that is what I'm doing.  I put the static code analysis all in one place...

File: src/test/java/com/amazonaws/encryptionsdk/internal/AwsKmsCmkArnInfoTest.java [hunk 1]
```diff
@@ -3,487 +3,442 @@
...
+
+      assertEquals(
+          AwsKmsCmkArnInfo.awsKmsArnMatchForDecrypt(
+              "c83fc47acc97", "64339c87-2ae4-42b1-8875-c83fc47acc97"),
+          false);
+    }
+
+    @Test
+    @DisplayName(
+        "Check for early return (Postcondition): Both identifiers are not ARNs and not equal, therefore they can not match.")
```
Comment (lavaleri) [inline src/test/java/com/amazonaws/encryptionsdk/internal/AwsKmsCmkArnInfoTest.java:361]: formatter breaks test conditions here

File: codebuild/openjdk11.yml [hunk 2]
```diff
@@ -6,4 +6,5 @@ phases:
             java: openjdk11
     build:
         commands:
+            - mvn com.coveo:fmt-maven-plugin:check
```
Comment (seebees) [inline codebuild/openjdk11.yml:9]:
```suggestion
```

I think we only need to do this in `static-analysis`?

File: codebuild/corretto11.yml [hunk 3]
```diff
@@ -6,4 +6,5 @@ phases:
             java: corretto11
     build:
         commands:
+            - mvn com.coveo:fmt-maven-plugin:check
```
Comment (seebees) [inline codebuild/corretto11.yml:9]:
```suggestion
```
I think we only need to do this in `static-analysis`?

**Review by seebees — APPROVED** [review]
Looks good.  Can you add the `jardiff` and `diff` commands you used for posterity :)

**Comment by lavaleri** [comment]
To check byte code difference:
- checkout this branch, `mvn clean`, `mvn install`
- `mv ./target/aws-encryption-sdk-java-2.3.0.jar ./after.jar` 
- checkout master branch, `mvn clean` `mvn install`
- `mv ./target/aws-encryption-sdk-java-2.3.0.jar ./before.jar`
- install jardiff: https://github.com/scala/jardiff
- `jardiff before.jar after.jar`
- confirm that any output is just pom or META-INF updates

To verify doc changes:
- checkout this branch, `mvn clean`, `mvn javadoc:javadoc`
- `mv target/site afterdocs`
- checkout master branch, `mvn clean`, `mvn javadoc:javadoc`
- `mv target/site beforedocs`
- strip newlines and `<!--Generatedbyjavadoc(11.0.10)onWedMayXXXX:XX:XXPDT2021-->` tags from both `afterdocs` and `beforedocs` using your favorite regex!
- `diff -r beforedocs afterdocs`
- confirm there is no output

**Review by alex-chew — APPROVED** [review]
LGTM

**Pull request merged**